### PR TITLE
Resin wall armor changes

### PR DESCRIPTION
**Pull request by Barnet2**
## About The Pull Request
Resin walls takes double damage from melee
Increases bullet and laser armor
For reference, a T60 (everyones favorite gun to mow down resin walls) will take 80 shots to down instead of 30.

## Why It's Good For The Game
As funny as aimmoding all the walls down, it isn't exactly healthy encouraging aim mode balls. Meleeing walls in this day and age is pointless unless you have a plasmacutter. The point of this PR is to reward taking more risk for more reward, as before there was no reason to take the risk of meleeing a wall. The melee increase is up to change if it is too much.

## Changelog

:cl:
balance: Resin walls take double damage from melee, but have highly increased bullet and laser armor
/:cl:

### CODE REVIEW

**Comment by Tonklaatech** [comment]
https://cdn.discordapp.com/attachments/498569646014857221/1206364967452803072/Locking-In.png?ex=65dbbded&is=65c948ed&hm=6e6422e239974351fac1bc9fbfb0aa35c1e9a61304d6d2c9d31a2edca3f5d5f0&

**Comment by bowlofcereal** [comment]
fortnite bowl

**Comment by MiniMeatwad** [comment]
T60 spam villains 3

**Comment by novaepee** [comment]
303906991-3edb0cea-fdc1-4150-a144-f35bf59f6955

**Comment by Lumipharon** [comment]
This makes it hideously difficult to shoot down walls **without** a whole gang of t60 dudes.

**Comment by WinterDarkraven** [comment]
is this a stealth t60 aimball buff chat?

**Comment by TellyLsc** [comment]
Clearing a mazes with a knife now only takes 200 registered melee clicks instead of 400! Wow!
Because God forbid there's a good autofire option to clear them.

**Review by Kuro020 — APPROVED** [review]
grenades and flamers still kill walls just as fast, stop crying about it

**Comment by TellyLsc** [comment]
> grenades and flamers still kill walls just as fast, stop crying about it

Grenades? You mean those weight 2 items that only do 80 damage at most in a less than 5x5 area? How many integrity do resin walls have again? Oh 150-300, if I'm reading resin.dm right.  That means it takes at least 2 nades to clear a wall assuming they're freshly built and hasn't matured to 300 integrity, this is not to mention the small AoE of HEDPs (and the lack of wallpierce) and how space inefficient grenades are unless you use the dedicated grenade storage/carry a box around.

And with flamers not having ext or spray nozzle anymore, well, that's gonna be annoying.
I'd give it to this PR though, a machete should on paper be able to destroy a single resin wall in at least 3-5 clicks with the extra damage, but hey, 3-5 clicks times 50 walls is still 150-250 clicks.

**Comment by Wisemonster** [comment]
Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.

**Comment by MiniMeatwad** [comment]
You're gonna see more of "Xeno builds resin wall in front of fob barricade, somehow doesn't explode, and melts barricade".

**Comment by TellyLsc** [comment]
> Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.

Yes, it wasn't, it took 2 sprays to clear a maze due to the weaker flames as well anyway. Which is even more baffling as to why it was deemed to be "too strong" to warrant being removed from the vendor.

**Comment by efzapa** [comment]
> > Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.
> 
> Yes, it wasn't, it took 2 sprays to clear a maze due to the weaker flames as well anyway which made it grossly wasteful in fuel usage even if it's the *most convenient* (not to be confused with effective) way to clear them. Which is even more baffling as to why it was deemed to be "too strong" to warrant being removed from the vendor.

fuel usage doesn't really mean shit when you can literally drop with 30 welder packs

**Comment by InterroLouis** [comment]
A hilariously massive buff to combat walls.

**Comment by TellyLsc** [comment]
> > > Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.
> > 
> > 
> > Yes, it wasn't, it took 2 sprays to clear a maze due to the weaker flames as well anyway which made it grossly wasteful in fuel usage even if it's the _most convenient_ (not to be confused with effective) way to clear them. Which is even more baffling as to why it was deemed to be "too strong" to warrant being removed from the vendor.
> 
> fuel usage doesn't really mean shit when you can literally drop with 30 welder packs for free

Yeah, let's just relegate all that clicking into the RO not knowing to hold spacebar to vend items and to alt+click tiles to place a conveyor belt under the vendors.

**Comment by efzapa** [comment]
> > > > Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.
> > > 
> > > 
> > > Yes, it wasn't, it took 2 sprays to clear a maze due to the weaker flames as well anyway which made it grossly wasteful in fuel usage even if it's the _most convenient_ (not to be confused with effective) way to clear them. Which is even more baffling as to why it was deemed to be "too strong" to warrant being removed from the vendor.
> > 
> > fuel usage doesn't really mean shit when you can literally drop with 30 welder packs for free
> 
> Yeah, let's just relegate all that clicking into the RO not knowing to hold spacebar to vend items and to alt+click tiles to place a conveyor belt under the vendors which makes up most of them.
> 
> Because let's increase the reliance to a "boring and repetitive" role on a game with poor/nonexistent tutorials, effectively increasing the skill floor.

clearly arguing in bad faith if you brought up an "RO role" in a discussion about sprays (which were removed from HvX entirely) and welding tanks (which you can get from the clothing vendor, like I said for free)

if you're gonna whine about a spray nerf, look into how the mechanics work before going on a rant about a role not even related to ANYTHING with said mechanic

**Comment by TellyLsc** [comment]
> > > > > Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.
> > > > 
> > > > 
> > > > Yes, it wasn't, it took 2 sprays to clear a maze due to the weaker flames as well anyway which made it grossly wasteful in fuel usage even if it's the _most convenient_ (not to be confused with effective) way to clear them. Which is even more baffling as to why it was deemed to be "too strong" to warrant being removed from the vendor.
> > > 
> > > 
> > > fuel usage doesn't really mean shit when you can literally drop with 30 welder packs for free
> > 
> > 
> > Yeah, let's just relegate all that clicking into the RO not knowing to hold spacebar to vend items and to alt+click tiles to place a conveyor belt under the vendors which makes up most of them.
> > Because let's increase the reliance to a "boring and repetitive" role on a game with poor/nonexistent tutorials, effectively increasing the skill floor.
> 
> clearly arguing in bad faith if you brought up an "RO role" in a discussion about sprays (which were removed from HvX entirely) and welding tanks (which you can get from the clothing vendor, like I said for free)
> 
> if you're gonna whine about a spray nerf, look into how the mechanics work before going on a rant about a role not even related to ANYTHING with said mechanic
> 
> (editing this as off topic after the fact so this thread doesn't get flooded but my point stands)

First of all, rude. Second of all, you were the one that brought up "spray nozz wasteful fuel usage isn't a concern because you can just vend 30 welder packs for free" and how are you going to get that welder pack to where it's actually relevant? At most you can store that welder pack crate in the Tad, but Tad obviously can't go into caves or be everywhere at once, which brings up the supply beacon which still can't get into caves but at least they're more portable, and once plopped down you'd still have to rely on who? The RO to fill up a crate of welding kits and send it to the correct beacon of course! And what, you want to dismiss all that because you can just personally vend 30 welding kits to alleviate the supply issue? Really? I think you're the one arguing in bad faith here.
Alright, maybe I'm understating the Tad relevance in transporting supplies but not everyone joins roundstart to have the chance to fill up a crate of stuff they care to bring into the Tad either so the point still stands.

Anyway that's the off-topic part done, now as for the resin wall armor changes I'd like to clarify that statistically this is a MASSIVE buff to walls against bullets, it's not just "+10 ranged soft armor and +15/10 hard ranged armor", it's a +50% DURABILITY increase for walls against projectiles.
Formerly a 25 damage, 5 AP bullet would inflict at least 8-9 damage per projectile to a resin wall? 
This effectively means that it can generally destroy a resin wall within 20 bullets.
With the addition of these extra armor it would theoretically withstand 30 bullets or so instead per wall, these numbers do add up when a screen's (of which is a 15x19 tile wide area) could generally be expected to contain somewhere between 25-50 tiles of resin walls, of course these estimates might be less on areas such as caves or buildings, but the nature of the indestructible cave walls and or the significantly more durable colony walls are not exactly in favour of the marines either if they do wish to push towards these chokepoints.
This buff isn't just "buffing resin walls to back when AP stat didn't work on walls" I think it goes way beyond that even. Which is absurd as a year ago the issue of "marines shooting the living crap out of mazes with MGs" weren't really a thing and we had railscope MG60s, antenna module req beacon, etc. which would've made it significantly easier than it would've been now if anything.

**Comment by MiniMeatwad** [comment]
I don't like combat walling, I don't like situations where you have to shoot 25-50% of your magazine before the wall goes down even when it was just built in your face.

In the past xenos would just resin wall in front of FOB and acid the cades, or even now you can reasonably easily combat wall to block marines from escaping or something. Lot easier to pull off with little downside since resin walls can be broken faster now.

Maybe a resin curing mechanic could improve this, like starting walls at 1 integrity, and giving them their expected integrity after 15 seconds.

**Comment by efzapa** [comment]
> > > > > > Spray wasn't used to clear walls, it was used to mass clear weeds while denying area.
> > > > > 
> > > > > 
> > > > > Yes, it wasn't, it took 2 sprays to clear a maze due to the weaker flames as well anyway which made it grossly wasteful in fuel usage even if it's the _most convenient_ (not to be confused with effective) way to clear them. Which is even more baffling as to why it was deemed to be "too strong" to warrant being removed from the vendor.
> > > > 
> > > > 
> > > > fuel usage doesn't really mean shit when you can literally drop with 30 welder packs for free
> > > 
> > > 
> > > Yeah, let's just relegate all that clicking into the RO not knowing to hold spacebar to vend items and to alt+click tiles to place a conveyor belt under the vendors which makes up most of them.
> > > Because let's increase the reliance to a "boring and repetitive" role on a game with poor/nonexistent tutorials, effectively increasing the skill floor.
> > 
> > clearly arguing in bad faith if you brought up an "RO role" in a discussion about sprays (which were removed from HvX entirely) and welding tanks (which you can get from the clothing vendor, like I said for free)
> > 
> > if you're gonna whine about a spray nerf, look into how the mechanics work before going on a rant about a role not even related to ANYTHING with said mechanic
> > 
> > (editing this as off topic after the fact so this thread doesn't get flooded but my point stands)
> 
> First of all, rude. Second of all, you were the one that brought up "spray nozz wasteful fuel usage isn't a concern because you can just vend 30 welder packs for free" and how are you going to get that welder pack to where it's actually relevant? At most you can store that welder pack crate in the Tad, but Tad obviously can't go into caves or be everywhere at once, which brings up the supply beacon which still can't get into caves but at least they're more portable, and once plopped down you'd still have to rely on who? The RO to fill up a crate of welding kits and send it to the correct beacon of course! And what, you want to dismiss all that because you can just personally vend 30 welding kits to alleviate the supply issue? Really? I think you're the one arguing in bad faith here.
> Alright, maybe I'm understating the Tad relevance in transporting supplies but not everyone joins roundstart to have the chance to fill up a crate of stuff they care to bring into the Tad either so the point still stands.
> 
> Anyway that's the off-topic part done, now as for the resin wall armor changes I'd like to clarify that statistically this is a MASSIVE buff to walls against bullets, it's not just "+10 ranged soft armor and +15/10 hard ranged armor", it's a +50% DURABILITY increase for walls against projectiles.
> Formerly a 25 damage, 5 AP bullet would inflict at least 8-9 damage per projectile to a resin wall? 
> This effectively means that it can generally destroy a resin wall within 20 bullets.
> With the addition of these extra armor it would theoretically withstand 30 bullets or so instead per wall, these numbers do add up when a screen's (of which is a 15x19 tile wide area) could generally be expected to contain somewhere between 25-50 tiles of resin walls, of course these estimates might be less on areas such as caves or buildings, but the nature of the indestructible cave walls and or the significantly more durable colony walls are not exactly in favour of the marines either if they do wish to push towards these chokepoints.
> This buff isn't just "buffing resin walls to back when AP stat didn't work on walls" I think it goes way beyond that even. Which is absurd as a year ago the issue of "marines shooting the living crap out of mazes with MGs" weren't really a thing and we had railscope MG60s, antenna module req beacon, etc. which would've made it significantly easier than it would've been now if anything.
> And while the addition of "increased melee damage multiplier" is a welcome one, the fact is, if a marine has the opportunity to clear a maze using their knife (or the machete if they are willing to bring it with how generally considered an awful opportunity cost it is), it is a time they're not just "moving along and helping the front of the push" as much as some people would regard "securing the rear so we don't get a maze built on our backs when we retreat" is an important thing, it's just often not as important as aiding the front of the push as to overwhelm xenos and win the round. 
> Why? Because this is ultimately a videogame, and people are interested in fighting the xenos, not "retreating to lick our wounds" or in the xeno's case "doing the delaylien strats" and extending the round needlessly for another back and forth slog until one side attritions enough, and if the main push fails and the marine wipes? Then that's it, the round's over, on to the next one.

tl;dr, cope

also you really wallposted about what I can tell, the TAD?

also dropping with said 30 welder packs isn't hard if you either bring them on alamo or have RO drop a crate of them which is literally one button lmao

you still have CAS, flames, AND nades like Kuro said

plus as Barnet said, MG60 bayo should 3-4 hit walls which is probably for the best to prevent "oh let's sit back 15 tiles and spray the hive with no consequence"

**Comment by Fawcksy** [comment]
"meleeing the walls was pointless"
I mean, I'm not complaining about this balance change, but... I've always just used my knife?

Im still trying to figure out why we axed flamer nozzles

**Comment by TiviPlus** [comment]
image

File: code/game/turfs/walls/resin.dm [hunk 1]
```diff
@@ -122,6 +123,8 @@
 	var/multiplier = 1
 	if(I.damtype == BURN) //Burn damage deals extra vs resin structures (mostly welders).
 		multiplier += 1
+	if(I.damtype == BRUTE) //Basically all melee attacks will do more damage.
+		multiplier += 0.5
```
Comment (TiviPlus) [inline code/game/turfs/walls/resin.dm:127]: 2 things: firstly this should be an else if
secondly the comment is wrong? remove it or reword it to explain this properly

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by mkvrs** [comment]
this sucks

**Comment by MiniMeatwad** [comment]
Return of the 3 hour round

**Comment by Looking4bros** [comment]
PC's are already the best Maze clearing tool, people just need to dedicate themselves to it. It would be a nice bump up for Machetes though. Two hitting most walls?
